nowcasting_dataset/data_sources/fake/utils.py

```diff
@@ -20,19 +20,25 @@ def join_list_data_array_to_batch_dataset(data_arrays: List[xr.DataArray]) -> xr
     return join_list_dataset_to_batch_dataset(datasets)
 
 
-def make_t0_datetimes_utc(batch_size):
+def make_t0_datetimes_utc(batch_size, temporally_align_batches: int = False):
     """
     Make list of t0 datetimes
 
     Args:
         batch_size: the batch size
+        temporally_align_batches: option to align batches in time
 
     Returns: pandas index of t0 datetimes
     """
 
     all_datetimes = pd.date_range("2021-01-01", "2021-02-01", freq="5T")
-    t0_datetimes_utc = np.random.choice(all_datetimes, batch_size, replace=False)
+
+    if temporally_align_batches:
+        t0_datetimes_utc = list(np.random.choice(all_datetimes, 1)) * batch_size
+    else:
+        t0_datetimes_utc = np.random.choice(all_datetimes, batch_size, replace=False)
```

Comment:
if we're sampling without replacement, then should we first check that `batch_size < len(all_datetimes)`?

Reply:
thanks, ive added something to stop thjis happening